components/processor/gcomapiprocessor/internal/gcom/client/client.go

```diff
@@ -0,0 +1,154 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/opentelemetry-collector-components/processor/gcomapiprocessor/internal/gcom/common"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	grafanaComReqs = promauto.NewHistogramVec(
```

Comment:
TODO: change this to otel, not sure if those would join the ones that are generated by the collector itself. In the future, the collector might not export automatically to Prometheus like it does today.